File: 0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public IList<int> InorderTraversal(TreeNode root) {
        List<int> ans=new List<int>();
        if(root!=null)
        DFS(root,ref ans);
        return ans;
    }
    private void DFS(TreeNode node,ref List<int> ans)
    {
        if(node==null)return;
        DFS(node.left,ref ans);
        ans.Add(node.val);
        DFS(node.right,ref ans);
    }
}

File: 0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> inorderTraversal(TreeNode* root) {
        vector<int> ans;
        Traversal(root,ans);
        return ans;
    }
    void Traversal(TreeNode* node,vector<int> & num){
        if(!node)return;
        Traversal(node->left,num);
        num.push_back(node->val);
        Traversal(node->right,num);
    }
};